src/controllers/admin/mentor.controller.ts

```diff
@@ -67,3 +68,33 @@ export const getAllMentorsByStatus = async (
     }
   }
 }
+
+export const getAllMentorEmails = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const user = req.user as Profile
+    const status: ApplicationStatus | undefined = req.query.status as
+      | ApplicationStatus
+      | undefined
+
+    if (user.type !== ProfileTypes.ADMIN) {
+      res.status(403).json({ message: 'Only Admins are allowed' })
+    } else {
+      if (status && !(status?.toUpperCase() in ApplicationStatus)) {
+        res.status(400).json({ message: 'Please provide a valid status' })
+        return
+      }
+      const { emails, statusCode, message } = await findAllMentorEmails(status)
+      res.status(statusCode).json({ emails, message })
+    }
+  } catch (err) {
+    if (err instanceof Error) {
+      console.error('Error executing query', err)
+      res
+        .status(500)
+        .json({ error: 'Internal server error', message: err.message })
+    }
+  }
+}
```

Comment:
We have refactored the code with return types. Can you take a pull and refer to an existing controller and update this? @sathudeva7